### PR TITLE
Fix/move to common build

### PR DESCRIPTION
**Pull request by Mpdreamz**
This moves the build over to the common build pattern in nullean, ecs-dotnet, elastic-transport.

Moves the SDK over to .NET 5.0

This now no longer hides `System.Reactive`, the build had to be run under windows for ILRepack to work and properly resign. Dependency free is nice but not worth spending weeks over trying to get there.

This also bumps the base TFM to net461 this falls in line with other repositories

### CODE REVIEW

**Comment by Mpdreamz** [comment]
The dependency listing turned out be a bug in `npkg-validator` https://github.com/nullean/nupkg-validator/pull/5

File: dotnet-tools.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  "isRoot": true,
+  "tools": {
+    "minver-cli": {
+      "version": "2.3.1",
+      "commands": [
+        "minver"
+      ]
+    },
+    "release-notes": {
+      "version": "0.2.2",
```
Comment (stevejgordon) [inline dotnet-tools.json]: Could now use 0.3.0

File: dotnet-tools.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      ]
+    },
+    "nupkg-validator": {
+      "version": "0.4.0",
+      "commands": [
+        "nupkg-validator"
+      ]
+    },
+    "assembly-differ": {
+      "version": "0.11.2",
```
Comment (stevejgordon) [inline dotnet-tools.json]: Could now use 0.13.0

**Review by stevejgordon — CHANGES_REQUESTED** [review]
A couple of the packages can be bumped to our latest releases but otherwise LGTM.

**Comment by Mpdreamz** [comment]
Good catch @stevejgordon updated the tooling to latest too.

**Review by stevejgordon — APPROVED** [review]
LGTM

**Pull request merged**